packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -0,0 +1,242 @@
+          return;
+        }
+        break;
+      case ' ':
+        // Space shouldn't trigger onAction so early return.
+
+        return;
+      case 'Home':
+      case 'End':
+      case 'PageDown':
+      case 'PageUp':
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        // Prevent these keys from moving the text cursor in the input
+        e.preventDefault();
+        // Move virtual focus into the wrapped collection
+        let focusCollection = new CustomEvent(FOCUS_EVENT, {
+          cancelable: true,
+          bubbles: true
+        });
+
+        collectionRef.current?.dispatchEvent(focusCollection);
+        break;
+      }
+      case 'ArrowLeft':
+      case 'ArrowRight':
+        // TODO: will need to special case this so it doesn't clear the focused key if we are currently
+        // focused on a submenutrigger? May not need to since focus would
+        // But what about wrapped grids where ArrowLeft and ArrowRight should navigate left/right
+        clearVirtualFocus();
+        break;
+    }
+
+    // Emulate the keyboard events that happen in the input field in the wrapped collection. This is for triggering things like onAction via Enter
+    // or moving focus from one item to another
+    if (state.focusedNodeId == null) {
+      collectionRef.current?.dispatchEvent(
+        new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+      );
+    } else {
+      let item = collectionRef.current?.querySelector(`#${CSS.escape(state.focusedNodeId)}`);
+      item?.dispatchEvent(
+        new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+      );
+    }
+  };
+
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-aria/autocomplete');
+  let collectionProps = useLabels({
+    id: collectionId,
+    'aria-label': stringFormatter.format('collectionLabel')
+  });
+
+  let {contains} = useFilter({sensitivity: 'base'});
+  let filterFn = useCallback((nodeTextValue: string) => {
+    if (defaultFilter) {
+      return defaultFilter(nodeTextValue, state.inputValue);
+    }
+
+    return contains(nodeTextValue, state.inputValue);
+  }, [state.inputValue, defaultFilter, contains]) ;
+
+  return {
+    inputProps: {
+      value: state.inputValue,
+      onChange: (e: ChangeEvent<HTMLInputElement>) => state.setInputValue(e.target.value),
+      onKeyDown,
+      autoComplete: 'off',
+      'aria-haspopup': 'listbox',
+      'aria-controls': collectionId,
+      // TODO: readd proper logic for completionMode = complete (aria-autocomplete: both)
+      'aria-autocomplete': 'list',
+      'aria-activedescendant': state.focusedNodeId ?? undefined,
+      // This disable's iOS's autocorrect suggestions, since the autocomplete provides its own suggestions.
+      autoCorrect: 'off',
+      // This disable's the macOS Safari spell check auto corrections.
+      spellCheck: 'false'
+    },
+    collectionProps: mergeProps(collectionProps, {
+      // TODO: shouldFocusOnHover?
```

Comment:
wonder if it might make sense to change focus on hover by default for the wrapped collection